onnx/defs/controlflow/defs.cc

```diff
@@ -263,6 +263,23 @@ void IfInferenceFunction(InferenceContext& ctx) {
 
       UnionShapeInfo(
           else_output->tensor_type().shape(), *if_output->mutable_tensor_type());
+    } else if (then_output->has_sequence_type() && then_output->sequence_type().elem_type().has_tensor_type()) {
```

Comment:
Could we move this logic (the complete if-then-else) as a function, say "UnionTypeInfo(const TypeProto& type1, TypeProto& type2)" into the shared utility file https://github.com/onnx/onnx/blob/master/onnx/defs/shape_inference.h ? It will help if the same logic is required elsewhere and if we generalize this in the future to handle map-types, etc. as well.

Reply:
The if-then-else in lines 237 to 245 should also be merged into the same function. The code here needs to just say
```cpp
   *if_output = *then_output;
   UnionTypeInfo(*else_output, *if_output);
```

Reply:
Thanks, updated